module/mempool/queue/chunkDataPack.go

```diff
@@ -0,0 +1,132 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/mempool"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+)
+
+// ChunkDataPackRequestMessageStore implements a HeroCache-based in-memory queue for storing chunk data pack requests.
+// It is designed to be utilized at Execution Nodes to maintain and respond chunk data pack requests.
+type ChunkDataPackRequestMessageStore struct {
+	mu        sync.RWMutex
+	cache     *herocache.Cache
+	sizeLimit uint
+}
+
+// Put enqueues the chunk data pack request message into the message store.
+// It expects the message.OriginID to be the requester identifier, and
+// message.Payload to be a chunk identifier.
+//
+// Boolean returned variable determines whether enqueuing was successful, i.e.,
+// put may be dropped if queue is full or already exists.
+func (c *ChunkDataPackRequestMessageStore) Put(message *engine.Message) bool {
+	return c.push(message.Payload.(flow.Identifier), message.OriginID)
+}
+
+// Get pops the queue, i.e., it returns the head of queue, and updates the head to the next element.
+// Boolean return value determines whether pop is successful, i.e., poping an empty queue returns false.
+func (c *ChunkDataPackRequestMessageStore) Get() (*engine.Message, bool) {
+	head, ok := c.pop()
+	if !ok {
+		return nil, false
+	}
+	return &engine.Message{
```

Comment:
From my perspective it would be ideal if the Herocache-backed implementation of `MessageStore` was generic to message type (like the corresponding [FifoQueue](https://github.com/onflow/flow-go/blob/e1099a88efe6ad90c0afb8e4679990e6f97857c4/engine/fifoqueue.go#L7-L22)). Also hopefully a fruitful place to apply generics, like Alexey mentioned. 

Out of scope for this PR, just noting for consideration when/if we make use of Herocache for a message queue in a different context, in the future.

Reply:
Noted here: https://github.com/dapperlabs/flow-go/issues/6362#issuecomment-1227880978